brightwind/analyse/analyse.py

```diff
@@ -2245,3 +2252,295 @@ def calc_air_density(temperature: Union[float, pd.Series, pd.DataFrame],
+    air temperature by the relative humidity.
+       
+    For more detailed references see the following:
+    https://wahiduddin.net/calc/density_altitude.htm
+    https://help.emd.dk/knowledgebase/content/ReferenceManual/DensityOfAir.pdf
+
+    
+    :param relative_humidity_percent:   Relative humidity values expressed as a percentage (%).
+                                        Represents the ratio of actual water vapour pressure to the 
+                                        saturation vapour pressure at the same temperature.
+    :type relative_humidity_percent:    float or pandas.Series
+    :param air_temperature_degC:        Air temperature in degrees Celsius.
+    :type air_temperature_degC:         float or pandas.Series
+    :return:                            Actual vapour pressure in hectopascals (hPa).
+                                        Output type depends on input type. If all inputs are float, 
+                                        output is float. If any input is pandas.Series, 
+                                        output is pandas.Series.
+    :rtype:                             float or pandas.Series
+
+        **Calculation method:**
+        The vapour pressure (P_v) is calculated using the relationship between relative humidity 
+        and the saturation vapour pressure at the given temperature:
+
+            P_v = (RH / 100) x E_s(T)
+
+        where:
+            P_v   = actual vapour pressure (hPa)  
+            RH    = relative humidity (%)  
+            E_s(T) = saturation vapour pressure (hPa) at temperature T (°C),  
+                     computed using `calc_saturation_vapour_pressure_of_water_vapour()`.
+
+    **Example usage**
+    ::
+    import brightwind as bw
+
+    # Calculate vapour pressure for 60% relative humidity at 20°C
+    bw.calc_vapour_pressure_from_relative_humidity(60.0, 20.0)
+    # 14.023424867988654 hPa
+
+    # Apply to a pandas Series of temperature and relative humidity data
+    DATA = bw.load_csv(bw.demo_datasets.demo_data)
+    DATA = bw.apply_cleaning(DATA, bw.demo_datasets.demo_cleaning_file)
+
+    bw.calc_vapour_pressure_from_relative_humidity(DATA['RH2m'], DATA['T2m']).tail(5)
+
+    # Timestamp
+    # 2017-11-23 10:10:00    6.519788
+    # 2017-11-23 10:20:00    6.377994
+    # 2017-11-23 10:30:00    6.473225
+    # 2017-11-23 10:40:00    6.511171
+    # 2017-11-23 10:50:00    6.473225
+    # dtype: float64
+    """
+    if isinstance(relative_humidity_percent, pd.Series) or isinstance(air_temperature_degC, pd.Series):
+        if len(relative_humidity_percent) != len(air_temperature_degC):
+            raise ValueError("relative_humidity_percent and air_temperature_degC must have the same dimensions.") 
+    P_v_hPa = (relative_humidity_percent/100)*calc_saturation_vapour_pressure_of_water_vapour(air_temperature_degC)
+    return P_v_hPa
+
+
+def calc_vapour_pressure_from_dewpoint(dew_point_temperature_degC):
```

Comment:
this is exactly same function as `calc_saturation_vapour_pressure_of_water_vapour`, if we want to implement method I was suggesting for calc_air_density and have all other functions private then I don't think that we need this function